dotCMS/src/main/java/com/dotcms/storage/RequestMetadata.java

```diff
@@ -0,0 +1,123 @@
+package com.dotcms.storage;
+
+import com.dotmarketing.util.UtilMethods;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Encapsulates the parameters to request the metadata
+ * it could be by cache or file system
+ * @author jsanca
+ */
+public class RequestMetadata {
+
+    /**
+     * Provides the key for the storage
+     */
+    private final StorageKey    storageKey;
+
+    /**
+     * If true, means the medatada output will be stores in the memory cache.
+     */
+    private final boolean           cache;
+
+    /**
+     * Cache key supplier, if cache is true
+     */
+    private final Supplier<String>  cacheKeySupplier;
+
+    /**
+     * In case the metadata is retrieved from the storage instead of the cache,
+     * you can wrap the metadata recovery from the storage in order to add, mod or remove values
+     */
+    private final Function<Map<String, Serializable>, Map<String, Serializable>> wrapMetadataMapForCache;
+
+    /**
+     * Builder based constructor
+     * @param builder
+     */
+    private RequestMetadata(final Builder builder) {
+        this.cache                   = builder.cache;
+        this.cacheKeySupplier        = builder.cacheKeySupplier;
+        this.storageKey              = builder.storageKey;
+        this.wrapMetadataMapForCache = builder.wrapMetadataMapForCache;
+    }
+
+    public StorageKey getStorageKey() {
+        return storageKey;
+    }
+
+    /**
+     * cache setting read
+     * @return
+     */
+    public boolean isCache() {
+        return cache;
+    }
+
+    public Supplier<String> getCacheKeySupplier() {
+        return cacheKeySupplier;
+    }
+
+    public Function<Map<String, Serializable>, Map<String, Serializable>> getWrapMetadataMapForCache() {
+        return wrapMetadataMapForCache;
+    }
+
+    /**
+     * Convenience Builder
+     */
+    public static final class Builder {
+
+        /**
+         * Provides the key for the storage
+         */
+        private  StorageKey    storageKey;
```

Comment:
Codacy found an issue: [Field storageKey has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)